comit/src/float_math.rs

```diff
@@ -0,0 +1,223 @@
+use bitcoin::hashes::core::cmp::Ordering;
+use num::{BigUint, Zero};
+use std::str::FromStr;
+
+/// Truncate the float's mantissa to length `precision`.
+pub fn truncate(float: f64, precision: u16) -> f64 {
+    let s = float.to_string();
+    truncate_str(&s, precision)
+}
+
+/// Truncate the float's mantissa to length `precision`.
+pub fn truncate_str(float: &str, precision: u16) -> f64 {
```

Comment:
I think it would make more sense for it to return a string as it took a string slice as input.  the conversion back in a float can be done in `truncate`.

```suggestion
pub fn truncate_str(number: &str, precision: u16) -> f64 {
```

Reply:
Aggree.